File: android-app/app/src/main/java/com/shop/android/models/Product.kt
package com.shop.android.models

import com.google.gson.annotations.SerializedName

/**
 * مدل داده محصول
 * مطابق با API بک‌اند Go
 */
data class Product(
    @SerializedName("id")
    val id: Int,
    
    @SerializedName("name")
    val name: String,
    
    @SerializedName("description")
    val description: String,
    
    @SerializedName("price")
    val price: Double,
    
    @SerializedName("images")
    val images: List<String>,
    
    @SerializedName("category")
    val category: String,
    
    @SerializedName("brand")
    val brand: String,
    
    @SerializedName("stock")
    val stock: Int,
    
    @SerializedName("sku")
    val sku: String? = null,
    
    @SerializedName("rating")
    val rating: Double? = null,
    
    @SerializedName("reviews_count")
    val reviewsCount: Int? = null
) 

File: app/types/product.ts
export interface Product {
  id: number
  sku?: string
  slug?: string
  name: string
  description?: string
  full_description?: string
  meta_description?: string
  seo_title?: string
  price: number
  image?: string
  category?: string | {
    name: string
    slug: string
  }
  rating?: number
  stock?: number
  stock_quantity?: number
  in_stock?: boolean
  brand?: string
  discount?: number
  isNew?: boolean
  isFeatured?: boolean
  disable_buy_button?: boolean
  call_for_price?: boolean
  isSpecial?: boolean
  isFavorite?: boolean
} 